modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -128,11 +128,6 @@ module ChapelIteratorSupport {
   }
 
   inline proc _getIteratorZip(type t: _tuple) {
-    if (!(isTupleType(t))) then
-      compilerError("cannot zip over a type that is not a homogeneous tuple");
-    else if (!(isEnumType(t(1)))) then
-      compilerError("cannot zip over a tuple of types that is not of enums");
-    else {
```

Comment:
You may want to fix the indentation in the code that follows as a result of removing the if/else statements